pytorch_lightning/tuner/tuning.py

```diff
@@ -119,47 +113,88 @@ def scale_batch_size(
                 - ``model.hparams``
                 - ``model.datamodule``
                 - ``trainer.datamodule`` (the datamodule passed to the tune method)
-
-            **fit_kwargs: remaining arguments to be passed to .fit(), e.g., dataloader
-                or datamodule.
-
         """
-        self.setup_trainer(model, **fit_kwargs)
-        return scale_batch_size(
-            self.trainer,
+        self.trainer.auto_scale_batch_size = True
+        result = self.trainer.tune(
             model,
-            mode,
-            steps_per_trial,
-            init_val,
-            max_trials,
-            batch_arg_name,
-            **fit_kwargs,
+            train_dataloader=train_dataloader,
+            val_dataloaders=val_dataloaders,
+            datamodule=datamodule,
+            scale_batch_size_kwargs={
+                'mode': mode,
+                'steps_per_trial': steps_per_trial,
+                'init_val': init_val,
+                'max_trials': max_trials,
+                'batch_arg_name': batch_arg_name,
+            }
         )
+        self.trainer.auto_scale_batch_size = False
+        return result['scale_batch_size']
 
     def lr_find(
         self,
         model: 'pl.LightningModule',
         train_dataloader: Optional[DataLoader] = None,
         val_dataloaders: Optional[Union[DataLoader, List[DataLoader]]] = None,
+        datamodule: Optional['pl.LightningDataModule'] = None,
         min_lr: float = 1e-8,
         max_lr: float = 1,
         num_training: int = 100,
         mode: str = 'exponential',
         early_stop_threshold: float = 4.0,
-        datamodule: Optional['pl.LightningDataModule'] = None,
         update_attr: bool = False,
     ) -> Optional[_LRFinder]:
-        self.setup_trainer(model, train_dataloader, val_dataloaders, datamodule)
-        return lr_find(
-            self.trainer,
+        """
+        Enables the user to do a range test of good initial learning rates,
+        to reduce the amount of guesswork in picking a good starting learning rate.
+
+        Args:
+            model: Model to tune.
+
+            train_dataloader: A Pytorch DataLoader with training samples. If the model has
+                a predefined train_dataloader method this will be skipped.
+
+            val_dataloaders: Either a single Pytorch Dataloader or a list of them, specifying validation samples.
+                If the model has a predefined val_dataloaders method this will be skipped
+
+            datamodule: An instance of :class:`~pytorch_lightning.core.datamodule.LightningDataModule`.
+
+            min_lr: minimum learning rate to investigate
+
+            max_lr: maximum learning rate to investigate
+
+            num_training: number of learning rates to test
```

Comment:
Not related to this PR. `num_training` param doesn't seem intuitive.

Reply:
cc: @SkafteNicki 

I'm just moving the docstring around here